tool/utils.ts

```diff
@@ -56,146 +57,107 @@ interface ReleaseInfo {
+ *
+ * By default, downloads the release version specified in package.json. Throws
+ * if an error occurs.
  */
-export async function getDartSassEmbedded(options: {
-  outPath: string;
-  version?: string;
-  path?: string;
-  release?: boolean;
-}): Promise<void> {
+export async function getDartSassEmbedded(
+  outPath: string,
+  options?:
+    | {
+        version: string;
+      }
+    | {
+        ref: string | 'main';
+      }
+    | {
+        path: string;
+      }
+): Promise<void> {
   const repo = 'dart-sass-embedded';
 
-  if (options.release) {
-    const release = options.version
-      ? {tag_name: options.version, name: `sass_embedded ${options.version}`}
-      : await getLatestReleaseInfo({
-          repo,
-        });
+  if (!options || 'version' in options) {
+    const version = options?.version ?? pkg['compiler-version'];
     await downloadRelease({
       repo,
       assetUrl:
         `https://github.com/sass/${repo}/releases/download/` +
-        `${release.tag_name}/` +
-        `${release.name.replace(' ', '-')}-` +
-        `${OS}-${ARCH}` +
-        ARCHIVE_EXTENSION,
-      outPath: options.outPath,
+        `${version}/sass_embedded-${version}-` +
+        `${OS}-${ARCH}${ARCHIVE_EXTENSION}`,
+      outPath,
     });
-    fs.rename(
-      p.join(options.outPath, 'sass_embedded'),
-      p.join(options.outPath, repo)
-    );
-  } else if (options.path) {
-    buildDartSassEmbedded(options.path);
-    await linkBuiltFiles(
-      p.join(options.path, 'build'),
-      p.join(options.outPath, repo)
-    );
-  } else {
+    fs.rename(p.join(outPath, 'sass_embedded'), p.join(outPath, repo));
+    return;
+  }
+
+  if ('ref' in options) {
     fetchRepo({
       repo,
       outPath: BUILD_PATH,
-      ref: options.version,
+      ref: options.ref,
     });
-    buildDartSassEmbedded(p.join(BUILD_PATH, repo));
-    await linkBuiltFiles(
-      p.join(BUILD_PATH, repo, 'build'),
-      p.join(options.outPath, repo)
-    );
   }
-}
 
-// Gets the ReleaseInfo of the latest release for `repo`. If `version` is given,
-// throws an error if the latest version is not semver-compatible with
-// `version`. If `tag` is true, gets the latest tag instead of release.
-async function getLatestReleaseInfo(options: {
-  repo: string;
-  versionConstraint?: string;
-  tag?: boolean;
-}): Promise<ReleaseInfo> {
-  console.log(`Getting version info for ${options.repo}.`);
-  const response = await fetch(
-    'https://api.github.com/repos/sass/' +
-      `${options.repo}/${options.tag ? 'tags' : 'releases'}`,
-    {
-      redirect: 'follow',
-    }
-  );
-  if (!response.ok) {
-    throw Error(
-      `Failed to get version info for ${options.repo}: ${response.statusText}`
-    );
-  }
-  const latestRelease = JSON.parse(await response.text())[0];
-  const latestVersion = options.tag
-    ? latestRelease.name
-    : latestRelease.tag_name;
+  const repoPath = 'path' in options ? options.path : p.join(BUILD_PATH, repo);
```

Comment:
what is the use case for this, as fetching does not respect `options.path` ?

Reply:
If `path` is provided, the compiler will be built from the source at that path. If `ref` is provided, the compiler will be built from the source at that Github ref.